### PR TITLE
Added test for create vm celery task

### PR DESCRIPTION
**Pull request by pcoder**
This PR includes a test case that helps testing the celery `create_vm_task` on the production.

We need to set the `STRIPE_API_PRIVATE_KEY_TEST` .env variable to test stripe key as this is what is used in the test.

### CODE REVIEW

File: datacenterlight/tests.py [hunk 1]
```diff
@@ -1,3 +1,120 @@
...
+        # We create a VM from the first template available to DCL
+        vm_template = VMTemplate.objects.all().first()
+        template_data = VMTemplateSerializer(vm_template).data
+
+        # The specs of VM that we want to create
+        specs = {
+            'cpu': 1,
+            'memory': 2,
+            'disk_size': 10,
+            'price': 15,
```
Comment (telmich) [inline datacenterlight/tests.py]: Should price still be in there? Should it not always be calculated? (even in a test)

File: datacenterlight/tests.py [hunk 2]
```diff
@@ -1,3 +1,120 @@
...
+
+        stripe_customer = StripeCustomer.get_or_create(
+            email=self.customer_email,
+            token=self.token)
+        billing_address = BillingAddress(
+            cardholder_name=self.customer_name,
+            postal_code='1232',
+            country='CH',
+            street_address='Monty\'s Street',
+            city='Hollywood')
```
Comment (telmich) [inline datacenterlight/tests.py]: I'd prefer Linthal - but nvm

File: datacenterlight/tests.py [hunk 3]
```diff
@@ -1,3 +1,120 @@
...
+        billing_address.save()
+        billing_address_data = {'cardholder_name': self.customer_name,
+                                'postal_code': '1231',
+                                'country': 'CH',
+                                'token': self.token,
+                                'street_address': 'Monty\'s Street',
+                                'city': 'Hollywood'}
```
Comment (telmich) [inline datacenterlight/tests.py]: Or Elm

File: datacenterlight/tests.py [hunk 4]
```diff
@@ -1,3 +1,120 @@
...
+
+        billing_address_id = billing_address.id
+        vm_template_id = template_data.get('id', 1)
+        final_price = specs.get('price')
```
Comment (telmich) [inline datacenterlight/tests.py]: And here we are -- I suggest to calculate, so we never ever take a price verbatim (more a style than logic question)

File: datacenterlight/tests.py [hunk 5]
```diff
@@ -1,3 +1,120 @@
...
+
+        # Make stripe charge to a customer
+        stripe_utils = StripeUtils()
+        charge_response = stripe_utils.make_charge(
+            amount=final_price,
+            customer=stripe_customer.stripe_id)
+
+        # Check if the payment was approved
+        if not charge_response.get(
```
Comment (telmich) [inline datacenterlight/tests.py]: Should this be and'ed? What if not charge_response.get('response_object') but charge_response.get('paid')? ... Wait. If we don't have a response_object, we don't even need to check for paid?

I am a bit confused by this logic

File: datacenterlight/tests.py (hunk 1 above)
Comment (pcoder) [inline datacenterlight/tests.py]: Yes, it should be. Unfortunately, to be consistent we our current code that expects price in there, we need to pass it.
We have two issues in redmine for refactoring this #3730 and #3736 which will hopefully fix this.

File: datacenterlight/tests.py (hunk 2 above)
Comment (verysanghee) [inline datacenterlight/tests.py]: -_- @telmich

File: datacenterlight/tests.py (hunk 5 above)
Comment (pcoder) [inline datacenterlight/tests.py]: Removed the unnessary statement `charge_response.get('paid')`

**Pull request merged**